contrib/Profiler/Profiler.py

```diff
@@ -0,0 +1,337 @@
+            min_gpu = sample_list[i]._gpu_usage
+            min_gpu_idx = i
+        if sample_list[i]._gpu_usage > max_gpu:
+            max_gpu = sample_list[i]._gpu_usage
+            max_gpu_idx = i
+        if sample_list[i]._mem_used < min_mem:
+            min_mem = sample_list[i]._mem_used
+            min_mem_idx = i
+        if sample_list[i]._mem_used > max_mem:
+            max_mem = sample_list[i]._mem_used
+            max_mem_idx = i
+        if sample_list[i]._read_bytes > max_read:
+            max_read = sample_list[i]._read_bytes
+            max_read_idx = i
+
+        if sample_list[i]._gpu_usage <= 10:
+            cpu_when_gpu_low.append(sample_list[i]._cpu_usage)
+            mem_when_gpu_low.append(sample_list[i]._mem_used)
+            disk_read_when_gpu_low.append(sample_list[i]._read_bytes)
+
+        sum_cpu += sample_list[i]._cpu_usage
+        sum_gpu += sample_list[i]._gpu_usage
+        sum_mem += sample_list[i]._mem_used
+        sum_read += sample_list[i]._read_bytes
+
+    length_gpu_low = len(cpu_when_gpu_low)
+    if length_gpu_low == 0:
+        length_gpu_low = 1
+
+    print("%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f" %
+          (max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+           sum(cpu_when_gpu_low) / length_gpu_low,
+           sum(mem_when_gpu_low) / length_gpu_low,
+           sum(disk_read_when_gpu_low) / length_gpu_low))
+    return [max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+            sum(cpu_when_gpu_low) / length_gpu_low,
+            sum(mem_when_gpu_low) / length_gpu_low,
+            sum(disk_read_when_gpu_low) / length_gpu_low]
+
+
+def Start_Sample(container_id, container_pid, period, flag, one_duration, dir, gpu_id):
+    if not os.path.exists('./' + dir):
+        os.mkdir(dir)
+    realtime_log = csv.writer(open('./' + dir + '/log_result.csv', 'w', newline=''))
+    analyze_log = csv.writer(open('./' + dir + '/analyze_result.csv', 'w', newline=''))
+
+    str_write_realtime = ['cpu_usage', 'mem_used', 'IO_read', 'IO_write', 'network_receive', 'network_transmit']
+    for i in range(len(gpu_id)):
+        str_write_realtime.append('gpu_usage_' + str(gpu_id[i]))
+        str_write_realtime.append('gpu_mem_used_' + str(gpu_id[i]))
+    realtime_log.writerow(str_write_realtime)
+
+    analyze_log.writerow(['max_gpu', 'avg_gpu', 'max_cpu', 'avg_cpu', 'max_mem', 'avg_mem', 'max_read', 'sum_read',
+                          'avg_cpu_gpu_low', 'avg_mem_gpu_low', 'avg_IO_gpu_low'])
+    nvmlInit()
+    sample_list = list()
+    Container_CPU_filelist = list()
+    Container_MEM_filelist = list()
+    Container_BLK_filelist = list()
+    Container_NET_file = ''
+
+    print(
+        "max_gpu\tavg_gpu\tmax_cpu\tavg_cpu\tmax_memory\tavg_memory\tmax_read\ttotal_read\tavg_cpu_when_gpu_low\tavg_mem_when_gpu_low\tavg_io_when_gpu_low")
+
+    if flag == 0:
```

Comment:
could you pass a `host_type` string here and check whether it is `"host"` or `"Docker"`, `flag` is meaningless and unnecessary?